File: medium/stack.py
import sys


class Stack(object):
    numbers = []

    def push(self, num):
        self.numbers.append(num)

    def pop(self):
        return self.numbers.pop()


if __name__ == "__main__":
    f = open(sys.argv[1])
    s = Stack()
    for l in f:
        nums = l.rstrip().split(' ')
        to_print = []
        for n in nums:
            s.push(n)
        i = 0
        while(True):
            try:
                n = s.pop()
            except IndexError:
                break
            if i % 2 is 0:
                to_print.append(n)
            i += 1
        print ' '.join(to_print)

File: medium/stack.rb
class Stack
  def initialize
    @e = []
  end

  def push(e)
    @e.push(e)
  end

  def pop
    @e.pop
  end
end

File.readlines(ARGV[0]).each do |l|
  the_stack = Stack.new
  l.split.each { |i| the_stack.push(i) }
  result = []

  loop do
    break if (e = the_stack.pop).nil?
    result.push(e)
    the_stack.pop
  end
  puts result.join(' ')
end
